src/ol/expr/cpu.js

```diff
@@ -183,6 +183,9 @@ function compileExpression(expression, context) {
     case Ops.Interpolate: {
       return compileInterpolateExpression(expression, context);
     }
+    case Ops.Coalesce: {
+      return compileCoalesceExpression(expression, context);
```

Comment:
I pushed a commit that moved this case up with the string and number type assertions.